pkg/server/storage_api/storage_api_test: TestAdminDecommissionedOperations failed

pkg/server/storage_api/storage_api_test.TestAdminDecommissionedOperations [failed](https://tanzanite.cluster.engflow.com/invocations/default/44e651b0-1d54-46e5-b2e7-691d2fe67430?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL3N0b3JhZ2VfYXBpOnN0b3JhZ2VfYXBpX3Rlc3Q=) on master @ [a5c82bcd4a34cd150eccb83b80d1fd70f94bf91d](https://github.com/cockroachdb/cockroach/commits/a5c82bcd4a34cd150eccb83b80d1fd70f94bf91d):


```
=== RUN   TestAdminDecommissionedOperations
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestAdminDecommissionedOperations3636996286
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #81590 (expected label: C-bug)
    decommission_test.go:977: -- test log scope end --
test logs left over in: outputs.zip/logTestAdminDecommissionedOperations3636996286
--- FAIL: TestAdminDecommissionedOperations (2.86s)
=== RUN   TestAdminDecommissionedOperations/Events
    decommission_test.go:972: 
        	Error Trace:	pkg/server/storage_api/storage_api_test_test/pkg/server/storage_api/decommission_test.go:972
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            				github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:42
        	            				pkg/server/storage_api/storage_api_test_test/pkg/server/storage_api/decommission_test.go:954
        	Error:      	Not equal: 
        	            	expected: 0xd
        	            	actual  : 0x7
        	Test:       	TestAdminDecommissionedOperations/Events
        	Messages:   	rpc error: code = PermissionDenied desc = admin-events: RangeIterator failed to seek to /Table/12/2: failed to connect to n1 at 127.0.0.1:36289: initial connection heartbeat failed: grpc: n2 was permanently removed from the cluster at 2024-06-20 07:55:37.759841492 +0000 UTC; it is not allowed to rejoin the cluster [code 7/PermissionDenied], node is decommissioned
    --- FAIL: TestAdminDecommissionedOperations/Events (0.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminDecommissionedOperations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39666

--- a/pkg/server/storage_api/decommission_test.go
+++ b/pkg/server/storage_api/decommission_test.go
@@ -889,7 +889,7 @@ func TestAdminDecommissionedOperations(t *testing.T) {
 			})
 			return err
 		}},
-		{"Events", codes.Internal, func(ctx context.Context, c serverpb.AdminClient) error {
+		{"Events", codes.PermissionDenied, func(ctx context.Context, c serverpb.AdminClient) error {
 			_, err := c.Events(ctx, &serverpb.EventsRequest{})
 			return err
 		}},

--- a/pkg/sql/colexecerror/error.go
+++ b/pkg/sql/colexecerror/error.go
@@ -214,18 +214,17 @@ func shouldCatchPanic(panicEmittedFrom string) bool {
 // stack, such as the network or storage layers. A StorageError will be bubbled
 // up all the way past the SQL layer unchanged.
 type StorageError struct {
-	error
+	cause error
 }
 
-// Cause implements the Causer interface.
-func (s *StorageError) Cause() error {
-	return s.error
-}
+func (s *StorageError) Error() string { return s.cause.Error() }
+func (s *StorageError) Cause() error  { return s.cause }
+func (s *StorageError) Unwrap() error { return s.cause }
 
 // NewStorageError returns a new storage error. This can be used to propagate
 // an error through the exec subsystem unchanged.
 func NewStorageError(err error) *StorageError {
-	return &StorageError{error: err}
+	return &StorageError{cause: err}
 }
 
 // notInternalError is an error that occurs not because the vectorized engine

